service/internal/access/v2/pdp.go

```diff
@@ -227,6 +245,111 @@ func (p *PolicyDecisionPoint) GetDecision(
 	return decision, nil
 }
 
+func (p *PolicyDecisionPoint) GetDecisionRegisteredResource(
+	ctx context.Context,
+	registeredResourceValueFQN string,
+	action *policy.Action,
+	resources []*authz.Resource,
+) (*Decision, error) {
+	l := p.logger.With("registeredResourceValueFQN", registeredResourceValueFQN)
+	l = l.With("action", action.GetName())
+	l.DebugContext(ctx, "getting decision", slog.Int("resourcesCount", len(resources)))
+
+	if err := validateGetDecisionRegisteredResource(registeredResourceValueFQN, action, resources); err != nil {
+		return nil, err
```

Comment:
Do you think we should wrap any additional context around these errors?

Reply:
That method returns explicit errors, so not sure we need any more context.